packages/agent-toolkit/src/core/tools/platform-api-tools/create-doc-tool.ts

```diff
@@ -0,0 +1,200 @@
+import { z } from 'zod';
+
+import {
+  createDoc as createDocMutation,
+  addContentToDocFromMarkdown,
+  getItemBoard,
+  createColumn as createColumnMutation,
+  updateDocName,
+} from '../../../monday-graphql/queries.graphql';
+import {
+  BoardKind,
+  ColumnType,
+  CreateColumnMutation,
+  CreateColumnMutationVariables,
+  CreateDocMutation,
+  CreateDocMutationVariables,
+  GetItemBoardQuery,
+  GetItemBoardQueryVariables,
+  UpdateDocNameMutation,
+  UpdateDocNameMutationVariables,
+  AddContentToDocFromMarkdownMutation,
+  AddContentToDocFromMarkdownMutationVariables,
+} from '../../../monday-graphql/generated/graphql';
+import { ToolInputType, ToolOutputType, ToolType } from '../../tool';
+import { BaseMondayApiTool, createMondayApiAnnotations } from './base-monday-api-tool';
+
+// Create discriminated union for document location
+const CreateDocLocationSchema = z.discriminatedUnion('type', [
+  z.object({
+    type: z.literal('workspace').describe('Create document in workspace'),
+    workspace_id: z.number().describe('Workspace ID under which to create the new document'),
+    doc_kind: z.nativeEnum(BoardKind).optional().describe('Document kind (public/private/share). Defaults to private.'),
```

Comment:
Why default to private, in monday usually stuff are created publicly by default no?